tests/installation/accept_license.pm

```diff
@@ -25,12 +25,14 @@ use version_utils qw(is_sle sle_version_at_least);
 sub run {
     my ($self) = @_;
 
-    # workaround for bsc#1069124: Skip the license check in upgrade mode
-    # During upgrade to sle 15, license agreement is shown on aarch64 and s390x
-    # but not shown on ppc64le and x86_64 at the moment
-    if (is_sle && sle_version_at_least(15) && get_var('UPGRADE') && !check_var('ARCH', 'aarch64') && !check_var('ARCH', 's390x')) {
-        send_key $cmd{next};
-        return record_soft_failure('bsc#1069124: License is not shown in upgrade');
+    # During upgrade to sle 15, license agreement is shown as soon as possible
+    # when we know which product to upgrade:
+    # - s390x and aarch64 installation media contain always just one product,
+    #   so the license agreement is shown at welcome screen
+    # - other architectures contain more products, the license agreement won't
+    #   be shown until user chooses which product to upgrade
+    if (is_sle && sle_version_at_least(15) && get_var('UPGRADE')) {
+        return send_key $cmd{next} unless check_screen('license-agreement', 5);
```

Comment:
I recommend to combine this with the next assert_screen two lines below, add the tag based on the if-clause and use match_has_tag as is done below the assert_screen anyway, e.g:
```perl
my @tags = qw(network-settings-button license-agreement);
push @tags, 'license-agreement' if is_sle && sle_version_at_least(15) && get_var('UPGRADE');
assert_screen \@tags;
if (match_has_tag 'network-settings-button') {
   ...
}
if (match_has_tag 'license-agreement') {
    send_key $cmd{next};
}
...
```

or maybe even as is done in tests/x11/kontact.pm with a `do {} until` loop.

Reply:
Yes, replace check_screen with multi-tag assert_screen would same some time during test. But it makes the test procedure a little more complicated. Please check my update.